sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathAccessControlExtensions.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace Azure.Storage.Files.DataLake.Models
+{
+    /// <summary>
+    /// Extension methods for RolePermissions.
+    /// </summary>
+    public static class PathAccessControlExtensions
+    {
+        /// <summary>
+        /// Parses octal char to RolePermissions.
+        /// </summary>
+        public static RolePermissions ParseOctalRolePermissions(char octalRolePermission)
+        {
+            RolePermissions rolePermissions = RolePermissions.None;
+
+            int value = (int)char.GetNumericValue(octalRolePermission);
+
+            if (value < 0 || value > 7)
+            {
+                throw Errors.MustBeBetweenInclusive(nameof(octalRolePermission), 0, 7, value);
+            }
+
+            if ((value & 4) > 0)
+            {
+                rolePermissions |= RolePermissions.Read;
+            }
+
+            if ((value & 2) > 0)
+            {
+                rolePermissions |= RolePermissions.Write;
+            }
+
+            if ((value & 1) > 0)
+            {
+                rolePermissions |= RolePermissions.Execute;
+            }
+
+            return rolePermissions;
+        }
+
+        /// <summary>
+        /// Parses symbolic permissions string to RolePermissions.
+        /// </summary>
+        /// <param name="symbolicRolePermissions">String to parse.</param>
+        /// <param name="allowStickyBit">If sticky bit is allowed.</param>
+        /// <returns><see cref="RolePermissions"/>.</returns>
+        public static RolePermissions ParseSymbolicRolePermissions(string symbolicRolePermissions, bool allowStickyBit)
+        {
+            RolePermissions rolePermissions = RolePermissions.None;
+            ArgumentException argumentException = new ArgumentException("Role permission contains an invalid character");
+
+            if (symbolicRolePermissions == null)
+            {
+                throw Errors.ArgumentNull(nameof(symbolicRolePermissions));
+            }
+
+            if (symbolicRolePermissions.Length != 3)
+            {
+                throw new ArgumentException("Role permission must be 3 characters");
```

Comment:
Nit - move to DataLakeErrors and include the actual string they're trying to parse.

Reply:
Added actual string, see my comment above.

Reply:
fixed.